influxql/ast.go

```diff
@@ -1296,14 +1296,27 @@ func (s *SelectStatement) String() string {
 }
 
 // RequiredPrivileges returns the privilege required to execute the SelectStatement.
-func (s *SelectStatement) RequiredPrivileges() ExecutionPrivileges {
-	ep := ExecutionPrivileges{{Admin: false, Name: "", Privilege: ReadPrivilege}}
+// NOTE: Statement must be normalized first (database name(s) in Sources and
```

Comment:
It's unintuitive, but the `QueryAuthorizer` code fills in the name using the default database if the name is an empty string. So since the default database gets filled in with the same value, they both end up getting the same database name when you don't specify one.

It's unintuitive which is why I thought it would be good to document it.